web-registry/src/hooks/useBasketDetails.ts

```diff
@@ -1,121 +1,235 @@
-        // TODO: Basket Curator
-        const curator = 'regen1qdqafsy2jfuyq7rxzkdwyytmrxlfn8csq0uetx';
-        const balances = await getBasketBalances(basketDenom);
-        const batches = await getBasketBalancesWithBatchInfo(balances.balances);
-
-        setData({
-          basket: basket.basket,
-          basketExtended: {
-            classes: basketClasses,
-            displayDenom,
-            description,
-            curator,
-            totalAmount: 0,
-          },
-          basketBalances: balances.balances,
-          batchesInfo: batches,
-        });
+        const _basketClasses = await Promise.all(
+          basketClassesInfo.map(async basketClass => {
+            // const metadata = await getMetadata(basketClass.info!.metadata);
+            return {
+              id: basketClass.info!.classId,
+              name: basketClass.info!.classId,
+            };
+          }),
+        );
+        setBasketClasses(_basketClasses);
       } catch (error) {
-        // TODO Manage error
         console.error(error); // eslint-disable-line no-console
       }
     }
 
-    fetchData(basketDenom);
-  }, [basketDenom]);
+    fetchData(basketClassesInfo);
+  }, [basketClassesInfo]);
 
-  // Prepare data for BasketOverview
+  // TODO ? creditaBatches data >> extract into its own hook / function ?
+  // fetch project data related to credit batch using graphql lazy hook
   useEffect(() => {
-    if (data?.basket && data?.basketExtended && data?.batchesInfo) {
-      const _dataOverview: BasketOverviewProps = {
-        name: data.basket.name,
-        displayDenom: data.basketExtended.displayDenom,
-        description: data.basketExtended.description,
-        totalAmount: data.basketExtended.totalAmount,
-        curator: data.basketExtended.curator,
-        allowedCreditClasses: data.basketExtended.classes,
-      };
-      if (data?.basket?.dateCriteria?.minStartDate) {
-        _dataOverview.minStartDate = data.basket.dateCriteria.minStartDate;
-      }
-      if (data?.basket?.dateCriteria?.startDateWindow) {
-        _dataOverview.startDateWindow =
-          data?.basket?.dateCriteria.startDateWindow;
-      }
+    if (!batches) return;
+
+    async function fetchData(batches: string[]): Promise<void> {
+      try {
+        const _batchesProjects = await Promise.all(
+          batches.map(async batchDenom => {
+            const batchProject = await fetchProjectByBatchDenom({
+              variables: { batchDenom },
+            });
 
-      if (data?.batchesInfo?.length > 0) {
-        _dataOverview.totalAmount = data.batchesInfo.reduce(
-          (acc: number, obj: BatchInfoWithProject) =>
-            acc + Number(obj.total_amount),
-          0,
+            // eslint-disable-next-line no-console
+            console.log('** batchProject', batchProject);
+
+            return {
+              batchDenom,
+              projectName:
+                batchProject.data!.creditVintageByBatchDenom!
```

Comment:
please use `?` instead of `!` because this can be null or undefined

Reply:
Perfect. Yesterday reviewing the models in graphql I realized that many fields of the project entity are not mandatory... I update the code.